### ISSUE TITLE
Can't get step definition to match when step contains nested literal parentheses

### ISSUE THREAD

**Issue by pmarflee** [cucumber/cucumber-js#1233]
I have the following scenario:
```
Scenario: Computing a point from a distance
  Given r <- ray(point(2, 3, 4), vector(1, 0, 0))
  Then  position(r, 0.0) = point(2.0, 3.0, 4.0)
    And position(r, 1.0) = point(3.0, 3.0, 4.0)
    And position(r, -1.0) = point(1.0, 3.0, 4.0)
    And position(r, 2.5) = point(4.5, 3.0, 4.0)
```
And the following step definitions (TypeScript):

```
Given("r <- ray\\(point\\({int}, {int}, {int}), vector\\({int}, {int}, {int}))", function (pX: number, pY: number, pZ: number, vX: number, vY: number, vZ: number) {
    return 'pending';
});

Then("position\\(r, {float}) = point\\({float}, {float}, {float})", function (t: number, x: number, y: number, z: number) {
    return 'pending';
});
```

When I run the tests, I receive the following warning:
```
1) Scenario: Computing a point from a distance # features\steps\rays.feature:10
   ? Given r <- ray(point(2, 3, 4), vector(1, 0, 0))
       Undefined. Implement with the following snippet:

         Given('r <- ray\(point\({int}, {int}, {int}), vector\({int}, {int}, {int}))', function (int, int2, int3, int4, int5, int6) {
           // Write code here that turns the phrase above into concrete actions
           return 'pending';
         });

   - Then position(r, 0.0) = point(2.0, 3.0, 4.0) # features\steps\raysSteps.ts:36
   - And position(r, 1.0) = point(3.0, 3.0, 4.0) # features\steps\raysSteps.ts:36
   - And position(r, -1.0) = point(1.0, 3.0, 4.0) # features\steps\raysSteps.ts:36
   - And position(r, 2.5) = point(4.5, 3.0, 4.0) # features\steps\raysSteps.ts:36
```

The `Given` step defintion is not being matched but the `Then` step is, despite the opening parentheses being escaped in the same way.  My guess is that the nested parentheses in the `Given` step is causing the problem.  Is there a workaround that would still allow me to use Cucumber expressions?

Node: v8.9.1
Cucumber.js: 5.1.0

**Comment by charlierudolph** [comment]
Thanks for the thorough writeup. Don't think there is a current workaround. Will open an issue in cucumber-expressions.

**Comment by aslakhellesoy** [comment]
I believe this is fixed in release 7.0.0. Please confirm.
